packages/slate/src/utils/path-utils.js

```diff
@@ -0,0 +1,201 @@
+import { List } from 'immutable'
+
+/**
+ * Compare paths `a` and `b` to see which is before or after.
+ *
+ * @param {List} a
+ * @param {List} b
+ * @return {Number|Null}
+ */
+
+function compare(a, b) {
+  a = a.toArray()
+  b = b.toArray()
```

Comment:
any reason we are calling `toArray` here? does it make the `for` iteration below much faster?

it seems we could at least do it after the check, `if (a.size !== b.size) return null` then coerce to make this faster in the base case.

Reply:
@jtadmor good call. Nope no reason. I actually had originally written the logic for arrays and then realized they needed to be lists instead and didn't feel like re-writing it 😄